hack/test-cmd.sh

```diff
@@ -74,6 +74,25 @@ function check-curl-proxy-code()
   echo "For address ${full_address}, got ${status} but wanted ${desired}"
   return 1
 }
+
+# TODO: remove this function when we do the retry inside the kubectl commands.
+function kubectl-with-retry()
+{
+  retry=true
+  count=0
+  ERROR_FILE="/tmp/kubectl-error"
```

Comment:
yeah, ensure-temp-dir uses mktemp, which will set KUBE_TEMP for you, guaranteed to be unique.